rest/main.go

```diff
@@ -146,3 +158,144 @@ func serverMainPersistentConfig(fs *flag.FlagSet, flagStartupConfig *StartupConf
 
 	return false, startServer(&sc, ctx)
 }
+
+func automaticConfigUpgrade(configPath string) (*StartupConfig, bool, error) {
+	legacyServerConfig, err := LoadServerConfig(configPath)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if legacyServerConfig.DisablePersistentConfig != nil && *legacyServerConfig.DisablePersistentConfig {
+		return nil, true, nil
+	}
+
+	startupConfig, dbConfigs, err := legacyServerConfig.ToStartupConfig()
+	if err != nil {
+		return nil, false, err
+	}
+
+	dbConfigs, err = sanitizeDbConfigs(dbConfigs)
+	if err != nil {
+		return nil, false, err
+	}
+
+	// Attempt to establish connection to server, add retry like its other use-case
```

Comment:
I guess outdated comment. The function does retry.

```suggestion
	// Attempt to establish connection to server
```